rfcs/0000-cells-commitments/ckb-cells-commitments.md

```diff
@@ -0,0 +1,146 @@
+---
+Number: "0000"
+Category: Standards Track
+Status: Proposal
+Author: Quake Wang <quake.wang@gmail.com>
+Created: 2023-08-14
+---
+
+# CKB Cells Commitments
+
+## Abstract
+
+This RFC describes a cell commitment format for CKB that can be used to verify the cell status in decentralized way.
+
+## Motivation
+
+In cross-chain or on-chain contract verification scenarios we often need to verify the state of a cell, e.g. to prove at which height a cell was created and at which height it was consumed. Currently, CKB has no way to provide such a commitment, and the only way to address this issue is through pegged blocks with a challenge mechanism or a centralized trust mechanism.
+
+This RFC proposes a cell commitment format to solve this issue in a decentralized, trustless and lightweight way.
+
+## Specification
+
+### Updatable Merkle Mountain Range
+
+We use an updatable Merkle Mountain Range ([MMR]) to store the cell status. The cell status is defined as a tuple of `(out_point, created_by, consumed_by)`, where `out_point` is the cell out point, `created_by` is the block number when the cell was created, and `consumed_by` is the block number when the cell was consumed. The `created_by` and `consumed_by` are both `u64` numbers and the `consumed_by` is set to `u64::MAX` if a cell is live.
```

Comment:
Is `out_point` serialized in molecule format?

Reply:
> Is `out_point` serialized in molecule format?

Yes, in line 27:
```
`out_point` is serialized as molecule binary format
```